Bootstrap 4 full support

djibe opened the pull request.

djibe committed:
Bootstrap 4 compatible

djibe committed:
Bootstrap 4 css

djibe committed:
Bootstrap 4 minified CSS

djibe committed:
Bootstrap 4 js

djibe committed:
Bootstrap 4 js minified

elamperti commented on README.md:
```diff
@@ -72,7 +72,7 @@ if (something) {
 | placement | 'bottom' | popover placement |
 | align | 'left' | popover arrow align |
-| donetext | '完成' | done button text |
+| donetext | 'Done' | done button text |
```
Has this default changed in the code as well?

djibe replied:
In BS4 code yes

elamperti commented on dist/bootstrap4-clockpicker.css:
```diff
@@ -0,0 +1,200 @@
+.clockpicker-plate {
+    background-color: #ededee;
+    border: 2px solid #dedede;
+	border-radius: 50%;
```
I see a mix of tabs and spaces here, you should unify those. Perhaps check what the project has been using and set an `.editorconfig` file while you are at it :)

djibe replied:
I'll do it

djibe replied:
Done

elamperti commented on dist/bootstrap4-clockpicker.js:
```diff
@@ -0,0 +1,881 @@
+ * ClockPicker v0.2.1 (http://weareoutman.github.io/clockpicker/)
+ * Copyright 2014 Wang Shenwei.
+ * Licensed under MIT (https://github.com/weareoutman/clockpicker/blob/gh-pages/LICENSE)
+ * Bootstrap 4 support by djibe
```
It should have been interesting to see the diff against the original source and I think this approach could bring several maintenance problems in the future (as bugs fixed here wouldn't be fixed in the other file, and/or features will change over time). 

Judging by this, I suggest you to just override the original js with this and just leave your `0.1` version for Bootstrap 4, and whoever wants to use it with Bootstrap 3 can use version `0.0.7` instead.

djibe replied:
Hi. I'm really a noob with GitHub so I don't understand anything about your requests. Sorry.
But don't hesitate to guide me to enhance it.
True I didn't test on mobile.
So you can test it here : https://www.diabeclic.com/daemonite/bootstrap-datetimepicker-bs4daemonite.html

elamperti replied:
Hi, no problem. My suggestions were mainly two things:
  * Unify spaces/tabs in the code (i.e. don't use tabs in one line, then spaces in the other; follow what the original code was using)
  * Instead of naming the file `bootstrap4-clockpicker.js`, just overwrite the previous one  (`bootstrap-clockpicker.js`). As yours could be a new version, then users who want to use the bootstrap 3 version can use something <= 0.0.7

Don't waste much time on this, as the owner of the repo is gone and PRs probably won't be merged anyway :(

Thanks for the demo link!

elamperti reviewed (changes requested):
Hey @djibe, I'm glad to see someone worked on Bootstrap 4 support! Have you tested it in mobile? Does it work well?

I left some comments regarding your PR. My main suggestion is to ditch Bootstrap 3 support altogether to make it easier to maintain (as old versions will still be available anyway).

It appears like @weareoutman isn't around anymore, so I don't know if we will be able to merge these changes anytime soon.

Anyhow, great work here! :)

djibe committed:
Update README.md

djibe committed:
Update bootstrap4-clockpicker.js

djibe committed:
Update bootstrap4-clockpicker.min.js

djibe committed:
Update bootstrap4-clockpicker.css

djibe committed:
Update bootstrap4-clockpicker.min.css

djibe committed:
Update README.md

djibe committed:
Update README.md

djibe committed:
v0.2

djibe committed:
v0.2

djibe committed:
Update README.md

djibe committed:
BS4 + material design

djibe committed:
Add files via upload

djibe committed:
Update README.md

djibe committed:
Add files via upload

djibe committed:
Delete clockpicker-bs4.png

djibe committed:
Add files via upload

djibe committed:
Update README.md

djibe committed:
Update README.md

djibe committed:
Update README.md

djibe committed:
Update README.md

djibe committed:
Update bower.json

djibe committed:
Delete bootstrap4-clockpicker.min.js

djibe committed:
0.2.1

djibe committed:
0.2.1

djibe committed:
Delete clockpicker-bs4 (2).png

djibe committed:
Delete clockpicker-bs4-material (2).png

djibe committed:
Delete clockpicker-bs4-material.png

djibe committed:
Delete clockpicker-bs4.png

djibe committed:
0.2.1

djibe committed:
0.2.1

djibe committed:
0.2.3

djibe committed:
0.2.2

djibe committed:
0.2.2

djibe committed:
0.2.2

djibe committed:
0.2.2

djibe committed:
0.2.2

poradz committed:
Removed this.parseInputValue() from getTime function

Parsing input value in getTime function broke picked value when beforeHide or beforeDone callbacks were in use.

Commit:
Prevented the popover position from overflowing outside the window

djibe committed:
Merge pull request #4 from poradz/poradz-position-fix

Prevented the popover position from overflowing outside the window

djibe committed:
Merge pull request #2 from poradz/poradz-callbacks-fix

Removed this.parseInputValue() from getTime function

djibe committed:
0.2.3

djibe committed:
Add files via upload

r3dqu33n committed:
Fix on am/pm parsing on load

djibe committed:
Merge pull request #5 from r3dqu33n/gh-pages

Fix on am/pm parsing on load, fix on cancel/ok button events

djibe committed:
0.2.3

rskrzypczak committed:
Fixed #8 (minify)

djibe committed:
Merge pull request #9 from rskrzypczak/gh-pages

Fixed #8 (minify)
